clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -699,6 +947,95 @@ public ConsumerRecords<K, V> poll(final long timeoutMs) {
         return poll(Duration.ofMillis(timeoutMs));
     }
 
+    private void sendFetches() {
+        if (!eventHandler.isEmpty()) {
+            final Optional<BackgroundEvent> backgroundEvent = eventHandler.poll();
```

Comment:
Maybe we leave the eventHandler.poll() at the poll() method, wdyt? Also I think we actually want to do eventHandler.drain() to get all the stuff in the queue.

Reply:
Done.